src/drawscreen.c

```diff
@@ -2342,21 +2337,7 @@ win_update(win_T *wp)
 			    ++new_rows;
 			else
 #endif
-			{
-#ifdef FEAT_DIFF
-			    if (l == wp->w_topline)
-			    {
-				int n = plines_win_nofill(wp, l, FALSE)
-								+ wp->w_topfill;
-				n -= adjust_plines_for_skipcol(wp);
-				if (n > wp->w_height)
-				    n = wp->w_height;
-				new_rows += n;
-			    }
-			    else
-#endif
-				new_rows += plines_win(wp, l, TRUE);
-			}
+			    new_rows += plines_correct_topline(wp, l, TRUE);
```

Comment:
The effect of `limit_winheight` is a bit different when using `plines_correct_topline()`, but it doesn't matter here, as when line height (including or excluding filler lines) exceeds `w_height`, the `new_rows > wp->w_height - row - 2` condition below will be true, and `new_rows` will be overwritten.